controllers/hvpa_controller.go

```diff
@@ -503,14 +502,14 @@ func (r *HvpaReconciler) scaleIfRequired(hpaStatus *autoscaling.HorizontalPodAut
 	default:
 		err := fmt.Errorf("TargetRef kind not supported %v in hvpa %v", kind, hvpa.Namespace+"/"+hvpa.Name)
 		log.Error(err, "Error")
-		return nil, nil, false, false, 0, nil, err
+		return nil, false, false, 0, nil, err
 	}
 
 	var desiredReplicas int32
-	if hpaStatus == nil {
+	if hvpa.Spec.Replicas == nil {
 		desiredReplicas = currentReplicas
 	} else {
-		desiredReplicas = hpaStatus.DesiredReplicas
+		desiredReplicas = *hvpa.Spec.Replicas
```

Comment:
What about hpaWeight?

Reply:
This just captures the HPA's recommendations. The actual calculations based on `hpaWeight` happens later in `getWeightedReplicas()`